src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -2413,4 +2419,246 @@ namespace SmartPeak
+      catch (const std::exception& e) {
+        LOGE << e.what();
+        return;
+      }
+      Parameter* use_smoothed_intensities = epd_param.findParameter("use_smoothed_intensities");
+      if (use_smoothed_intensities && use_smoothed_intensities->getValueAsString() == "true")
+      {
+        LOGW << "Without EPD, smoothing is not supported. Setting 'use_smoothed_intensities' to false!";
+        use_smoothed_intensities->setValueFromString("false");
+      }
+    }
+
+    //-------------------------------------------------------------
+    // configure and run feature finding
+    //-------------------------------------------------------------
+
+    OpenMS::FeatureMap feat_map;
+    std::vector< std::vector< OpenMS::MSChromatogram > > feat_chromatograms;
+    OpenMS::FeatureFindingMetabo ffmet;
+    parameters = ffmet.getParameters();
+    Utilities::updateParameters(parameters, epd_param);
+    ffmet.setParameters(parameters);
+    ffmet.run(m_traces_final, feat_map, feat_chromatograms);
+
+    size_t trace_count(0);
+    for (size_t i = 0; i < feat_map.size(); ++i)
+    {
+      if (!feat_map[i].metaValueExists("num_of_masstraces"))
+      {
+        LOGE << "MetaValue 'num_of_masstraces' missing from FFMetabo output!";
+        return;
+      }
+      trace_count += (size_t)feat_map[i].getMetaValue("num_of_masstraces");
+    }
+    if (trace_count != m_traces_final.size())
+    {
+      Parameter* remove_single_traces = ffm_param.findParameter("remove_single_traces");
+      if (remove_single_traces && remove_single_traces->getValueAsString() == "true")
+      {
+        LOGE << "FF-Metabo: Internal error. Not all mass traces have been assembled to features! Aborting.";
+        return;
+      }
+      else
+      {
+        LOGI << "FF-Metabo: " << (m_traces_final.size() - trace_count) << " unassembled traces have been removed.";
+      }
+    }
+
+    LOGI << "-- FF-Metabo stats --\n"
+      << "Input traces:    " << m_traces_final.size() << "\n"
+      << "Output features: " << feat_map.size() << " (total trace count: " << trace_count << ")";
+
+    // filter features with zero intensity (this can happen if the FWHM is zero (bc of overly skewed shape) and no peaks end up being summed up)
+    auto intensity_zero = [&](OpenMS::Feature& f) { return f.getIntensity() == 0; };
+    feat_map.erase(remove_if(feat_map.begin(), feat_map.end(), intensity_zero), feat_map.end());
+
+    // chromatograms
+    if (feat_chromatograms.size() == feat_map.size())
+    {
+      for (size_t i = 0; i < feat_chromatograms.size(); ++i)
+      {
+        for (size_t j = 0; j < feat_chromatograms[i].size(); ++j)
+        {
+          rawDataHandler_IO.getExperiment().addChromatogram(feat_chromatograms[i][j]);
+        }
+      }
+    }
+    else
+    {
+//      LOGE << "FF-Metabo: Internal error. The number of features (" << feat_chromatograms.size() << ") and chromatograms (" << feat_map.size() << ") are different! Aborting.";
+//      return;
```

Comment:
Should this be uncommented?

Reply:
yes it will be cleaned. this PR is actually a WIP. by the way, we are failing there when I execute the workflow at the moment. so it needs to be investigated.